model/pod/podutil.go

```diff
@@ -0,0 +1,105 @@
+package pod
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/pkg/errors"
+)
+
+// Constants representing default curl retry arguments.
+const (
+	curlDefaultNumRetries = 10
+	curlDefaultMaxSecs    = 100
+)
+
+// CurlCommand returns the command to curl the evergreen client and retries the request with the default retry parameters.
+func (p *Pod) CurlCommand(settings *evergreen.Settings) ([]string, error) {
+	var retryArgs string
+	if curlDefaultNumRetries != 0 && curlDefaultMaxSecs != 0 {
+		retryArgs = " " + curlRetryArgs(curlDefaultNumRetries, curlDefaultMaxSecs)
+	}
+	cmds, err := p.curlCommands(settings, retryArgs)
+	if err != nil {
+		return []string{}, errors.WithStack(err)
+	}
+	return []string{"CMD-SHELL", strings.Join(cmds, " && ")}, nil
+}
+
+// AgentCommand returns the arguments to start the agent.
+func (p *Pod) AgentCommand(settings *evergreen.Settings) []string {
+	return []string{
+		fmt.Sprintf("./%s", p.BinaryName()),
+		"agent",
+		fmt.Sprintf("--api_server=%s", settings.ApiUrl),
+		fmt.Sprintf("--mode=pod"),
+		fmt.Sprintf("--pod_id=%s", p.ID),
+		fmt.Sprintf("--pod_secret=%s", p.Secret),
+		fmt.Sprintf("--log_prefix=%s", filepath.Join(p.TaskContainerCreationOpts.WorkingDir, "agent")),
+		fmt.Sprintf("--working_directory=%s", p.TaskContainerCreationOpts.WorkingDir),
+		"--cleanup",
+	}
+}
+
+// ClientURL returns the URL used to get the latest Evergreen client version
+// directly from the Evergreen server.
+func (p *Pod) ClientURL(settings *evergreen.Settings) string {
+	return strings.Join([]string{
+		strings.TrimSuffix(settings.ApiUrl, "/"),
+		strings.TrimSuffix(settings.ClientBinariesDir, "/"),
+		p.ExecutableSubPath(),
+	}, "/")
+}
+
+// S3ClientURL returns the URL in S3 where the Evergreen client version can be
+// retrieved for this server's particular Evergreen build version.
+func (p *Pod) S3ClientURL(settings *evergreen.Settings) string {
+	return strings.Join([]string{
+		strings.TrimSuffix(settings.PodInit.S3BaseURL, "/"),
+		evergreen.BuildRevision,
+		p.ExecutableSubPath(),
+	}, "/")
+}
+
+// BinaryName returns the file name of the binary.
+func (p *Pod) BinaryName() string {
+	name := "evergreen"
+	if p.TaskContainerCreationOpts.OS == OSWindows {
+		return name + ".exe"
+	}
+	return name
+}
+
+// ExecutableSubPath returns the directory containing the compiled agents.
+func (p *Pod) ExecutableSubPath() string {
+	return filepath.Join(string(p.TaskContainerCreationOpts.Arch), p.BinaryName())
```

Comment:
I think this actually has to join both OS and arch with an underscore because binaries are stored in this path in S3 and in the app server:

The general format of the URL is this:
```go
clients/${GOOS}_${GOARCH}/evergreen (or evergreen.exe for Windows)
```
So I would include the `p.TaskContainerCreationOpts.OS` in the path.

Reply:
Sounds good, but some of the Archs already have the OS appended in front (ie. `evergreen.ArchWindowsAmd64` =  `windows_amd64`), so should I just add it for the Archs that don't have the OS specified?

Reply:
Yeah, the `evergreen.Arch` types are for distros specifically and the only allowed pod architectures are the ones in the DB model package for pods.